src/components/Comment.jsx

```diff
@@ -1,10 +1,27 @@
-export function Comment({ body, votes, author, created_at }) {
+import { deleteComment } from "../api";
+
+export function Comment({ body, votes, author, created_at, comment_id, setComments }) {
+  function handleDeleteComment(e) {
+    setComments((currentComments) => {
+      return currentComments.filter((comment) => {
+        return comment.comment_id !== comment_id;
+      });
+    });
+    deleteComment(comment_id).catch((e) => {
+      console.error(e);
+      window.alert("ERROR: Could not contact the server, try again later");
+    });
+  }
```

Comment:
Optimistic rendering is great here! You need to make sure however to bring the comment back if the deletion was unsuccessful in your `catch` block here. For this reason, you may opt to NOT tackle this optimistically.